src/components/forms/NetworkDevicesForm/edit/NetworkDevicePanel.tsx

```diff
@@ -0,0 +1,392 @@
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  ScrollableContainer,
+  SidePanel,
+  Spinner,
+  useNotify,
+} from "@canonical/react-components";
+import type { ReactNode } from "react";
+import { type FC } from "react";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import type { LxdNicDevice, LxdNoneDevice } from "types/device";
+import NotificationRow from "components/NotificationRow";
+import NetworkDeviceIPAddressEdit from "components/forms/NetworkDevicesForm/edit/NetworkDeviceIPAddressEdit";
+import { useNetworks } from "context/useNetworks";
+import { useProfiles } from "context/useProfiles";
+import NetworkSelector from "pages/projects/forms/NetworkSelector";
+import NetworkAclSelector from "pages/networks/forms/NetworkAclSelector";
+import {
+  getDeviceAcls,
+  isNoneDevice,
+  getExistingDeviceNames,
+} from "util/devices";
+import type { FormNetworkDevice } from "util/formDevices";
+import { deduplicateName } from "util/formDevices";
+import { getInheritedNetworks } from "util/configInheritance";
+import {
+  supportsNicDeviceAcls,
+  getNetworkAcls,
+  combineAcls,
+} from "util/networks";
+import usePanelParams, { panels } from "util/usePanelParams";
+import type { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+
+export interface NetworkDeviceFormValues {
+  name: string;
+  network: string;
+  acls?: string;
+  ipv4?: string;
+  ipv6?: string;
+}
+
+type NetworkDeviceMode =
+  | "create-local"
+  | "create-override"
+  | "edit-override"
+  | "edit-local";
+
+interface Props {
+  project: string;
+  formik: InstanceAndProfileFormikProps;
+  onSave?: () => void;
+}
+
+const NetworkDevicePanel: FC<Props> = ({
+  project,
+  formik: parentFormik,
+  onSave,
+}) => {
+  const notify = useNotify();
+  const panelParams = usePanelParams();
+  const { data: profiles = [], isLoading: profilesLoading } =
+    useProfiles(project);
+  const { data: networks = [] } = useNetworks(project);
+
+  const managedNetworks = networks.filter((network) => network.managed);
+
+  const existingDeviceNames = getExistingDeviceNames(
+    parentFormik.values,
+    profiles,
+  );
+  const inheritedNetworks = getInheritedNetworks(parentFormik.values, profiles);
+  const shouldDisplayIpAddresses =
```

Comment:
```suggestion
  const isInstance =
```
Maybe a bit clearer?